src/sugar.jl

```diff
@@ -412,13 +412,24 @@ julia> set((a=1, b=2), my_func, 100)
 """
 macro accessor(ex)
     def = splitdef(ex)
+    fname = def[:name]
     length(def[:args]) == 1 || throw(ArgumentError("@accessor only supports single argument functions. Overload `Accessors.set(obj, ::typeof($(def[:name])), v)` manually."))
     arg = only(def[:args])
     argname = splitarg(arg)[1]
     body_optic = MacroTools.replace(def[:body], argname, :_)
+    ftype = :(
+        if $fname isa Function
+            typeof($fname)
+        elseif $fname isa Type
+            Type{$fname}
+        else
+            # is it possible at all?
+            error("Unsupported accessor $(fname)::$(typeof($fname))")
+        end
```

Comment:
```suggestion
        if $fname isa Type
            Type{$fname}
        else
            typeof($fname)
        end
```
Not sure if this is better. Would this cover the case, `@accessor (f::MyStruct)(x) = ...`?